packages/teleterm/src/services/tshd/middleware.ts

```diff
@@ -91,14 +93,41 @@ export const withLogging = (logger: Logger) => {
         });
       },
 
-      sendMessage(message: any, next: Function) {
-        logger.info(`send: ${method}(${JSON.stringify(message.toObject())})`);
+      sendMessage(message, next) {
+        logger.info(
+          `send: ${method}(${JSON.stringify(
+            filterPasswords(message.toObject())
+          )})`
+        );
         next(message);
       },
     };
 
     return new grpc.InterceptingCall(nextCall(options), params);
   };
-
-  return interceptor;
 };
+
+function filterSensitiveProperties(
+  toFilter: object,
+  filteredProperties: string[]
+): object {
+  return transform(
```

Comment:
We don't send that many messages so it doesn't matter that much, but if the throughput was higher, it'd probably be better to mutate an existing object instead of creating a new one each time.

Reply:
Yes, I took that into account, but I agree that for now it doesn't matter that much